cmd/crictl/exec.go

```diff
@@ -0,0 +1,108 @@
+package main
+
+import (
+	"fmt"
+
+	"github.com/urfave/cli"
+	"golang.org/x/net/context"
+	pb "k8s.io/kubernetes/pkg/kubelet/api/v1alpha1/runtime"
+)
+
+var runtimeExecCommand = cli.Command{
+	Name:  "exec",
+	Usage: "exec a command in a container",
+	Flags: []cli.Flag{
+		cli.BoolFlag{
+			Name:  "sync",
+			Usage: "run a command in a container synchronously",
+		},
+		cli.StringFlag{
+			Name:  "id",
+			Value: "",
+			Usage: "id of the container",
+		},
+		cli.Int64Flag{
+			Name:  "timeout",
+			Value: 0,
+			Usage: "timeout for the command",
+		},
+		cli.BoolFlag{
+			Name:  "tty",
+			Usage: "exec a command in a tty",
+		},
+		cli.BoolFlag{
+			Name:  "stdin",
+			Usage: "stream stdin",
+		},
+	},
+	Action: func(context *cli.Context) error {
+		// Set up a connection to the server.
+		conn, err := getRuntimeClientConnection(context)
+		if err != nil {
+			return fmt.Errorf("failed to connect: %v", err)
+		}
+		defer conn.Close()
+		client := pb.NewRuntimeServiceClient(conn)
+
+		var opts = execOptions{
+			id:      context.String("id"),
+			timeout: context.Int64("timeout"),
+			tty:     context.Bool("tty"),
+			stdin:   context.Bool("stdin"),
+			cmd:     context.Args(),
+		}
+		if context.Bool("sync") {
+			err = ExecSync(client, opts)
+			if err != nil {
+				return fmt.Errorf("execing command in container synchronously failed: %v", err)
+			}
+			return nil
+		}
+		err = Exec(client, opts)
+		if err != nil {
+			return fmt.Errorf("execing command in container failed: %v", err)
+		}
+		return nil
+	},
+}
+
+// ExecSync sends an ExecSyncRequest to the server, and parses
+// the returned ExecSyncResponse.
+func ExecSync(client pb.RuntimeServiceClient, opts execOptions) error {
+	if opts.id == "" {
+		return fmt.Errorf("ID cannot be empty")
+	}
+	r, err := client.ExecSync(context.Background(), &pb.ExecSyncRequest{
+		ContainerId: opts.id,
+		Cmd:         opts.cmd,
+		Timeout:     opts.timeout,
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Println("Stdout:")
+	fmt.Println(string(r.Stdout))
+	fmt.Println("Stderr:")
+	fmt.Println(string(r.Stderr))
+	fmt.Printf("Exit code: %v\n", r.ExitCode)
+
+	return nil
+}
+
+// Exec sends an ExecRequest to server, and parses the returned ExecResponse
+func Exec(client pb.RuntimeServiceClient, opts execOptions) error {
+	if opts.id == "" {
+		return fmt.Errorf("ID cannot be empty")
+	}
+	r, err := client.Exec(context.Background(), &pb.ExecRequest{
+		ContainerId: opts.id,
+		Cmd:         opts.cmd,
+		Tty:         opts.tty,
+		Stdin:       opts.stdin,
+	})
+	if err != nil {
+		return err
+	}
+	fmt.Printf("Url: %v\n", r.Url)
```

Comment:
Yes, I think so. Could you check https://github.com/kubernetes/kubernetes/blob/master/pkg/client/unversioned/remotecommand/remotecommand.go to visit the URL?